sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java

```diff
@@ -974,22 +975,36 @@ protected void preClose() {
         super.preClose();
     }
 
+	private IoAcceptor createIoAcceptor(Factory<? extends IoHandler> handlerFactory) {
+		Session session = getSession();
+		FactoryManager manager = Objects.requireNonNull(session.getFactoryManager(), "No factory manager");
+		IoServiceFactory factory = Objects.requireNonNull(manager.getIoServiceFactory(), "No I/O service factory");
+		IoHandler handler = handlerFactory.create();
+		return factory.createAcceptor(handler);
+	}
+
+	private IoAcceptor getLocalIoAcceptor(Factory<? extends IoHandler> handlerFactory) {
+		if (localAcceptor == null) {
+			localAcceptor = createIoAcceptor(handlerFactory);
+		}
+		return localAcceptor;
+	}
+
+	private IoAcceptor getDynamicIoAcceptor(Factory<? extends IoHandler> handlerFactory) {
```

Comment:
Let's make these `protected` in accordance with our policy to enable users to easily override our implementation if they see fit.

Reply:
The original caller doBind is private too; must be changed to protected too to facilitate this overriding?

Reply:
> The original caller `doBind` is `private` too; must be changed to `protected` too to facilitate this overriding?

Yes